File: MusicManagerAPI/src/main/java/com/ldz/music/manager/model/converter/beans/user/music/status/bo/to/model/UserMusicStatusBOtoModel.java
package com.ldz.music.manager.model.converter.beans.user.music.status.bo.to.model;

import com.ldz.music.manager.model.UserMusicStatus;
import com.ldz.converter.container.ConverterContainer;
import com.ldz.converter.container.annot.BeanConverter;
import com.ldz.converter.container.inter.IConverter;
import com.ldz.music.manager.model.MusicType;
import com.ldz.music.manager.model.bo.UserMusicStatusBO;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Component;

import java.util.Set;
import java.util.stream.Collectors;

/**
 * Created by ldalzotto on 26/04/2017.
 */
@Component
@BeanConverter(initialBeanClass = UserMusicStatusBO.class, targetBeanClass = UserMusicStatus.class)
public class UserMusicStatusBOtoModel implements IConverter<UserMusicStatusBO, UserMusicStatus> {

    @Autowired ConverterContainer converterContainer;

    @Override public UserMusicStatus apply(UserMusicStatusBO userMusicStatusBO) {
        UserMusicStatus userMusicStatus = new UserMusicStatus();

        if (userMusicStatusBO.getUsername() != null) {
            userMusicStatus.setUsername(userMusicStatusBO.getUsername());
        }

        if (userMusicStatusBO.getMusicTypeBOs() != null) {
            Set<MusicType> musicTypes = userMusicStatusBO.getMusicTypeBOs()
                    .stream().map(musicTypeBO -> converterContainer.convert(musicTypeBO, MusicType.class))
                    .collect(Collectors.toSet());
            userMusicStatus.setMusicTypes(musicTypes);
        }

        return userMusicStatus;
    }
}

File: MusicManagerAPI/src/scala/com/ldz/music/manager/model/converter/bo/to/model/UserMusicStatusBOtoModel.scala
package com.ldz.music.manager.model.converter.bo.to.model

import com.ldz.converter.container.ConverterContainer
import com.ldz.converter.container.annot.BeanConverter
import com.ldz.converter.container.inter.IConverter
import com.ldz.music.manager.model.{MusicType, MusicTypeDTO, UserMusicStatus}
import com.ldz.music.manager.model.bo.{MusicTypeBO, UserMusicStatusBO}
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.stereotype.Component

import scala.collection.JavaConverters._

/**
  * Created by Loic on 20/06/2017.
  */
@Component
@BeanConverter(initialBeanClass = classOf[UserMusicStatusBO], targetBeanClass = classOf[UserMusicStatus])
class UserMusicStatusBOtoModel extends IConverter[UserMusicStatusBO, UserMusicStatus]{

  @Autowired
  private val converterContainer: ConverterContainer = null

  override def apply(userMusicStatusBO: UserMusicStatusBO): UserMusicStatus = {
    userMusicStatusBO match {
      case `userMusicStatusBO` if userMusicStatusBO != null => {
        var userMusicStatus = new UserMusicStatus

        //musictypes
        val musics = userMusicStatusBO.getMusicTypeBOs.map(a => converterContainer.convert(a, classOf[MusicType]))
        userMusicStatus.setMusicTypes(musics.asJava)
        userMusicStatus.setUsername(userMusicStatusBO.getUsername)
        userMusicStatus
      }
      case _ => null
    }
  }
}
